app/src/androidTest/java/com/android/bookswapp/ui/addbooktest.kt

```diff
@@ -0,0 +1,235 @@
+            ratingStr = "4",
+            photo = "https://example.com/photo.jpg",
+            Book_languageStr = "ENGLISH",
+            Isbn = "1234567890")
+
+    // Assert the book is created correctly
+    assertEquals("My Book", book?.title)
+    assertEquals("Author Name", book?.author)
+    assertEquals("This is a description", book?.description)
+    assertEquals(4, book?.rating)
+    assertEquals("https://example.com/photo.jpg", book?.photo)
+    assertEquals(BookLanguages.ENGLISH, book?.language)
+    assertEquals("1234567890", book?.isbn)
+  }
+
+  @Test
+  fun testCreateDataBook_InvalidData() {
+    // Test with invalid data (empty title)
+    var book =
+        createDataBook(
+            title = "",
+            author = "Author Name",
+            description = "This is a description",
+            ratingStr = "4",
+            photo = "https://example.com/photo.jpg",
+            Book_languageStr = "ENGLISH",
+            Isbn = "1234567890")
+
+    // Assert that the book is null due to invalid title
+    assertNull(book)
+
+    // Test with invalid rating
+    book =
+        createDataBook(
+            title = "My Book",
+            author = "Author Name",
+            description = "This is a description",
+            ratingStr = "invalid_rating",
+            photo = "https://example.com/photo.jpg",
+            Book_languageStr = "ENGLISH",
+            Isbn = "1234567890")
+
+    // Assert that the book is null due to invalid rating
+    assertNull(book)
+
+    // Test with invalid language
+    book =
+        createDataBook(
+            title = "My Book",
+            author = "Author Name",
+            description = "This is a description",
+            ratingStr = "4",
+            photo = "https://example.com/photo.jpg",
+            Book_languageStr = "INVALID_LANGUAGE",
+            Isbn = "1234567890")
+
+    // Assert that the book is null due to invalid language
+    assertNull(book)
+  }
+}
+
+// Simulating the listToBooksView interface implementation
+class MockListToBooksView(repository: Booksrepository) : listToBooksView(repository) {
+  var isBookAdded = false
+  var addedBook: DataBook? = null
+  var isBookDeleted = false
+  var isBookUpdated = false
+  var fetchedBooks: List<DataBook> = emptyList()
+
+  override fun Add_Book(book: DataBook) {
+    isBookAdded = true // Simulate adding the book
+    addedBook = book
+  }
+
+  fun Update_Book(book: DataBook) {
+    isBookUpdated = true // Simulate updating the book
+    addedBook = book
+  }
+
+  fun Delete_Book(book: DataBook) {
+    isBookDeleted = true // Simulate deleting the book
+    addedBook = book
+  }
+
+  fun Fetch_Books(books: List<DataBook>) {
+    fetchedBooks = books // Simulate fetching books
+  }
+}
+
+class MockBooksRepository : Booksrepository {
+  var isBookAdded = false
+  var isBookFetched = false
+  var isBookUpdated = false
+  var isBookDeleted = false
+  var shouldFail = false
+
+  override fun init(OnSucess: () -> Unit) {
```

Comment:
rename to onSucess, otherwise it's perfect !

Reply:
Now with the new app refactoring, see new [function method](https://github.com/BookswapEPFL/Bookswap/blob/main/app/src/main/java/com/android/bookswap/data/repository/BookRepository.kt)
These one are deprecated